Backend/src/main/java/com/example/backend/collaborations/service/CollaborationsService.java

```diff
@@ -54,14 +74,25 @@ public Collaboration addCollaboration(Long projectid, CollaborationDTO collabora
         return collaborationsRepository.save(collaboration);
     }
 
-    public Collaboration updateCollaboration(Long projectId, Long companyId, CollaborationDTO collaborationDTO) {
-        CollaborationId collaborationId = new CollaborationId(projectRepository.findById(projectId).get(), companyRepository.findById(companyId).get());
+    public Collaboration updateCollaboration(Long projectId, Long companyId, CollaborationDTO collaborationDTO, AppUser user) throws AuthenticationException, ProjectNotFoundException, CompanyNotFoundException, CollaborationNotFoundException, UserNotFoundException, ContactNotFoundException {
+        if (user == null) throw new AuthenticationException();
+
+        CollaborationId collaborationId;
+        if (projectRepository.findById(projectId).isPresent() && companyRepository.findById(companyId).isPresent()){
+            collaborationId = new CollaborationId(projectRepository.findById(projectId).get(), companyRepository.findById(companyId).get());
+        } else if (projectRepository.findById(projectId).isEmpty()){
```

Comment:
Trebali bi možda svi zajedno malo raspraviti o uvođenju dependency injectiona.

Reply:
Mozes objasniti sta si tu mislio, tj. sta bi tocno napravio

Reply:
Da, to sam malo loše rekao. Ovo je članak o dependensy injectiou: https://www.techtarget.com/searchapparchitecture/definition/dependency-injection. U praksi to znači da svugdje kada neki servis koristi repozitorij, ne bi koristio izravan poziv, nego imao interface. Kada se jednom napravi dosta ga je lagano koristiti, samo zahtjeva malo igranja s frameworkom i malo dodatne neurednosti koda kako sada sve(servisi i repozitoriji) u osnovi treba imati interface. Općenito je moderna praksa na velikim projektima zbog olakšavanja u testiranju, ponovnom korištenju koda i omogućuje da ljudi koji rade manje ovise jedni o drugom kodu, predlažem malo čitanja o tome.